### PR TITLE
feat: Adding Apache pinot support

### PR DESCRIPTION
**Pull request by wahab-io**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/data-on-eks/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

### Motivation

This is a work in progress for adding support for Apache Pinot. Functionality is complete, need time to write the blueprint documentation. Creating the PR to review `terraform` code.
Closes: #76 

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [x] Mandatory for new blueprints. Yes, I have added a example to support my blueprint PR
- [x] Mandatory for new blueprints. Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: distributed-databases/pinot/addons.tf [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+  upper   = false
+}
+
+resource "random_password" "sensitive_key" {
+  length  = 16
+  special = false
+}
+
+module "eks_data_addons" {
+  source = "github.com/wahab-io/terraform-aws-eks-data-addons?ref=v1.2.6"
```
Comment (vara-bonthu) [inline distributed-databases/pinot/addons.tf:168]: Change this to new tag release https://github.com/aws-ia/terraform-aws-eks-data-addons/releases/tag/v1.2.6

File: distributed-databases/pinot/addons.tf [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+        name  = "cluster.name"
+        value = local.cluster_name
+      },
+      {
+        name  = "controller.replicaCount"
+        value = 3
+      },
+      {
+        name  = "controller.persistence.storageClass"
+        value = "gp2"
```
Comment (vara-bonthu) [inline distributed-databases/pinot/addons.tf:190]: use gp3 since you are making this default

File: distributed-databases/pinot/data.tf [hunk 3]
```diff
@@ -0,0 +1,5 @@
+data "aws_eks_cluster_auth" "this" {
+  name = module.eks.cluster_name
+}
+
+data "aws_availability_zones" "available" {}
```
Comment (vara-bonthu) [inline distributed-databases/pinot/data.tf:5]: remove this file and move the code to top of `main.tf` file

File: distributed-databases/pinot/locals.tf [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#---------------------------------------------------------------
```
Comment (vara-bonthu) [inline distributed-databases/pinot/locals.tf:1]: remove this file and move this content to top of the `main.tf` file

File: distributed-databases/pinot/variables.tf [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+variable "region" {
+  description = "region"
+  type        = string
+}
+
+variable "eks_cluster_version" {
+  description = "EKS Cluster version"
+  type        = string
+  default     = "1.25"
```
Comment (vara-bonthu) [inline distributed-databases/pinot/variables.tf:15]: did you try with latest EKS version?

File: distributed-databases/pinot/variables.tf [hunk 6]
```diff
@@ -0,0 +1,34 @@
+variable "name" {
+  description = "Name of the VPC and EKS Cluster"
+  type        = string
+  default     = "pinot-on-eks"
+}
+
+variable "region" {
+  description = "region"
+  type        = string
```
Comment (vara-bonthu) [inline distributed-databases/pinot/variables.tf:9]: default value should be set to `us-west-2`

**Review by vara-bonthu — COMMENTED** [review]
Thanks for the PR @wahab-io 👍🏼 I left few minor comments

File: distributed-databases/pinot/addons.tf (hunk 1 above)
Comment (wahab-io) [reply distributed-databases/pinot/addons.tf]: Done

File: distributed-databases/pinot/data.tf (hunk 3 above)
Comment (wahab-io) [reply distributed-databases/pinot/data.tf]: Done

File: distributed-databases/pinot/locals.tf (hunk 4 above)
Comment (wahab-io) [reply distributed-databases/pinot/locals.tf]: Done

File: distributed-databases/pinot/variables.tf (hunk 5 above)
Comment (wahab-io) [reply distributed-databases/pinot/variables.tf:15]: Did some preliminary testing with 1.28 and seems to be working. Don't see anything specific to k8s version [here](https://docs.pinot.apache.org/operators/tutorials/deployment-pinot-on-kubernetes) or [here](https://docs.pinot.apache.org/basics/getting-started/public-cloud-examples/aws-quickstart) also no reported issues on k8s version.

File: distributed-databases/pinot/variables.tf (hunk 6 above)
Comment (wahab-io) [reply distributed-databases/pinot/variables.tf:9]: Done

File: distributed-databases/pinot/addons.tf (hunk 2 above)
Comment (wahab-io) [reply distributed-databases/pinot/addons.tf]: Done

File: distributed-databases/pinot/main.tf [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+  cluster_version = var.eks_cluster_version
+
+  vpc_id     = module.vpc.vpc_id
+  subnet_ids = module.vpc.private_subnets
+
+  cluster_endpoint_private_access = true # if true, Kubernetes API requests within your cluster's VPC (such as node to control plane communication) use the private VPC endpoint
+  cluster_endpoint_public_access  = true # if true, Your cluster API server is accessible from the internet. You can, optionally, limit the CIDR blocks that can access the public endpoint.
+
+  eks_managed_node_groups = {
+    pinot_node_group = {
```
Comment (vara-bonthu) [inline distributed-databases/pinot/main.tf:66]: Add a core nodegroup to deploy all the addons similar to other blueprints

File: distributed-databases/pinot/main.tf [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+      instance_types = ["m5.xlarge"]
```
Comment (vara-bonthu) [inline distributed-databases/pinot/main.tf:67]: Add labels and taints for pinot node group.
Ensure that you add tolerations and node selectors to the Pinot deployment

Comment (vara-bonthu) [reply distributed-databases/pinot/main.tf:67]: Let's use gp3 volumes with atleast 100 GB

File: distributed-databases/pinot/addons.tf [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+    main = {
+      provider_arn               = module.eks.oidc_provider_arn
+      namespace_service_accounts = ["kube-system:ebs-csi-controller-sa"]
+    }
+  }
+  tags = local.tags
+}
+
+#---------------------------------------------------------------
+# GP3 Encrypted Storage Class
```
Comment (vara-bonthu) [inline distributed-databases/pinot/addons.tf:19]: Disable gp2 as default

File: distributed-databases/pinot/addons.tf [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+    fsType    = "xfs"
+    encrypted = true
+    type      = "gp3"
+  }
+
+  depends_on = [kubernetes_annotations.gp2_default]
+}
+
+
+#---------------------------------------------------------------
```
Comment (vara-bonthu) [inline distributed-databases/pinot/addons.tf:57]: remove this

File: distributed-databases/pinot/addons.tf [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+  #---------------------------------------------------------------
+  eks_addons = {
+    aws-ebs-csi-driver = {
+      service_account_role_arn = module.ebs_csi_driver_irsa.iam_role_arn
+    }
+    coredns = {
+      preserve = true
+    }
+    vpc-cni = {
+      service_account_role_arn = module.vpc_cni_irsa.iam_role_arn
```
Comment (vara-bonthu) [inline distributed-databases/pinot/addons.tf:98]: remove this line

File: distributed-databases/pinot/main.tf (hunk 7 above)
Comment (wahab-io) [reply distributed-databases/pinot/main.tf]: Done

File: distributed-databases/pinot/main.tf (hunk 8 above)
Comment (wahab-io) [reply distributed-databases/pinot/main.tf:100]: Done

File: distributed-databases/pinot/addons.tf (hunk 9 above)
Comment (wahab-io) [reply distributed-databases/pinot/addons.tf:19]: Done

**Pull request merged**